microk8s-resources/actions/enable.cilium.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+set -e
+
+source $SNAP/actions/common/utils.sh
+
+if ! which helm 2>&1 >/dev/null; then
+	echo "Cilium installation requires helm. Please install via 'snap install helm'." >&2
+	exit 1
+fi
+
+ARCH=$(arch)
+if ! [ "${ARCH}" = "amd64" ]; then
+  echo "Cilium is not available for ${ARCH}" >&2
+  exit 1
+fi
+
+echo "Enabling Cilium"
+
+if [ ! -f "${SNAP_DATA}/bin/cilium" ]
+then
+  CILIUM_VERSION="master"
+  CILIUM_ERSION=$(echo $CILIUM_VERSION | sed 's/v//g')
+  CILIUM_DIR="cilium-$CILIUM_ERSION"
+  SOURCE_URI="https://github.com/cilium/cilium/archive"
+  CILIUM_CNI_CONF="plugins/cilium-cni/05-cilium-cni.conf"
+  CILIUM_LABELS="k8s-app=cilium"
+  NAMESPACE=kube-system
+
+  echo "Fetching cilium version $CILIUM_VERSION."
+  sudo mkdir -p "${SNAP_DATA}/tmp/cilium"
+  (cd "${SNAP_DATA}/tmp/cilium"
+  sudo "${SNAP}/usr/bin/curl" -L $SOURCE_URI/$CILIUM_VERSION.tar.gz -o "$SNAP_DATA/tmp/cilium/cilium.tar.gz"
+  sudo gzip -f -d "$SNAP_DATA/tmp/cilium/cilium.tar.gz"
+  sudo tar -xf "$SNAP_DATA/tmp/cilium/cilium.tar" "$CILIUM_DIR/install" "$CILIUM_DIR/$CILIUM_CNI_CONF")
+
+  sudo mv "$SNAP_DATA/args/cni-network/cni.conf" "$SNAP_DATA/args/cni-network/10-kubenet.conf" 2>/dev/null || true
+  sudo cp "$SNAP_DATA/tmp/cilium/$CILIUM_DIR/$CILIUM_CNI_CONF" "$SNAP_DATA/args/cni-network/05-cilium-cni.conf"
+
+  (cd "${SNAP_DATA}/tmp/cilium/$CILIUM_DIR/install/kubernetes"
+  helm template cilium \
+      --namespace $NAMESPACE \
+      --set global.cni.confPath="$SNAP_DATA/args/cni-network" \
+      --set global.cni.customConf=true \
+      --set global.containerRuntime.integration="containerd" \
+      --set global.containerRuntime.socketPath="$SNAP_COMMON/run/containerd.sock" \
+      | sudo tee cilium.yaml >/dev/null)
+
+  sudo mkdir -p "$SNAP_DATA/actions/cilium/"
+  sudo cp "$SNAP_DATA/tmp/cilium/$CILIUM_DIR/install/kubernetes/cilium.yaml" "$SNAP_DATA/actions/cilium.yaml"
+
+  "$SNAP/kubectl" "--kubeconfig=$SNAP/client.config" apply -f "$SNAP_DATA/actions/cilium.yaml"
+  "$SNAP/kubectl" "--kubeconfig=$SNAP/client.config" -n $NAMESPACE rollout status ds/cilium
+
+  # Fetch Cilium CLI binary and install
+  CILIUM_POD=$("$SNAP/kubectl" "--kubeconfig=$SNAP/client.config" -n $NAMESPACE get pod -l $CILIUM_LABELS -o json | jq -r '.items[0].metadata.name')
+  CILIUM_BIN=$(mktemp)
+  "$SNAP/kubectl" "--kubeconfig=$SNAP/client.config" -n $NAMESPACE cp $CILIUM_POD:/usr/bin/cilium $CILIUM_BIN >/dev/null
+  sudo mkdir -p "$SNAP_DATA/bin/"
+  sudo mv $CILIUM_BIN "$SNAP_DATA/bin/cilium"
+  sudo chmod +x "$SNAP_DATA/bin/"
+  sudo chmod +x "$SNAP_DATA/bin/cilium"
+
+  sudo rm -rf "$SNAP_DATA/tmp/cilium"
+fi
+
+echo "Restarting kube-apiserver"
+refresh_opt_in_config "allow-privileged" "true" kube-apiserver
```

Comment:
I think configuring the apiserver should come before trying to apply the manifest. Here is what i got:
```
$ microk8s.enable cilium
Enabling Cilium
Fetching cilium version master.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   121    0   121    0     0    614      0 --:--:-- --:--:-- --:--:--   614
100 21.8M  100 21.8M    0     0  45.5M      0 --:--:-- --:--:-- --:--:--  118M
configmap/cilium-config unchanged
serviceaccount/cilium unchanged
serviceaccount/cilium-operator unchanged
clusterrole.rbac.authorization.k8s.io/cilium unchanged
clusterrole.rbac.authorization.k8s.io/cilium-operator unchanged
clusterrolebinding.rbac.authorization.k8s.io/cilium unchanged
clusterrolebinding.rbac.authorization.k8s.io/cilium-operator unchanged
deployment.apps/cilium-operator configured
The DaemonSet "cilium" is invalid: 
* spec.template.spec.containers[0].securityContext.privileged: Forbidden: disallowed by cluster policy
* spec.template.spec.initContainers[0].securityContext.privileged: Forbidden: disallowed by cluster policy
Failed to enable cilium
``` 
That is mentioned here: https://gist.github.com/borkmann/84812fcac624fb659f66f0d514faa1c8#problems-deploying-cilium-daemon-set

Reply:
And we probably need a `microk8s.status --wait-ready` before trying to apply the yaml manifests.

Reply:
Thanks for trying this out, I'm migrating from an existing setup that I had working to this so it's easy for me to miss stuff like this. I'll reorder the apiserver configuration.